### PR TITLE
Improve get application related APIs

### PR DESCRIPTION
**Pull request by bhathiya**
### Proposed changes in this pull request

- Improve get-application related APIs

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java [hunk 1]
```diff
@@ -5294,11 +5256,34 @@ public void addURLTemplates(int apiId, API api, Connection connection) throws AP
...
+     * @param applicationName Name of the Application
+     * @param userId          Name of the User.
+     * @param groupId         Group ID
+     * @throws APIManagementException
+     */
+    public Application getApplicationWithOAuthApps(String applicationName, String userId, String groupId)
+            throws APIManagementException {
+
+        Application application = getApplicationByName(applicationName, userId, groupId);
+
```
Comment (praminda) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java]: Additional white spaces

Comment (bhathiya) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java]: are you referring to the new line? if yes, that was added for clarity. but can remove them if need. wdyt?

Comment (praminda) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java]: New lines before the for loop and return statement looks clean.
Additional new lines between method variable initialization looks bit noisy to me :)

Comment (bhathiya) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java]: Got it. Fixed it.

**Pull request merged**